lib/hooks/post.ts

```diff
@@ -18,7 +29,29 @@ export function useDeletePostMutation(postId: string) {
   return useMutation(() => axios.delete("/api/post/" + postId), {
     onSuccess: () => {
       queryClient.invalidateQueries("post-" + postId);
-      router.push('/');
+      router.asPath == "/account" ? router.reload() : router.push("/");
     },
   });
 }
+
+export interface EditablePostClient {
+  description?: string;
+  price?: number;
+  image?: Blob;
+}
+
+export function useEditPostMutation(postId: string) {
+  return useMutation((request: EditablePostClient) => {
+    const formData = new FormData();
+
+    if (request.description)
+      formData.append("description", request.description);
+
+    if (request.price)
+      formData.append("price", request.price.toFixed(2).replace(".", ""));
```

Comment:
This feels like something that should be take care of by the caller and instead just taking a number so as to match the API as closely as possible.

Reply:
I think I may not be understanding this comment. It's not possible to always pass in a number for price because if someone is editing a post, they may not changing the price. Is that what you are referring to?

Reply:
I just mean taking the price as an int number instead of a string

Reply:
That's not how the api is setup though. FormData only takes a blob or a string, and the api endpoint changes the string to a number upon request retrieval. You can see this here: https://github.com/calebmech/Book-Bazar/blob/main/pages/api/post/%5Bid%5D.ts#L185

Reply:
Keeping this unresolved for now. Let me know your thoughts.